lib/json/json.c

```diff
@@ -714,7 +673,38 @@ ssize_t json_escape(char *str, size_t *len, size_t buf_size)
 		return -ENOMEM;
 	}
 
-	return json_escape_buf(str, len, escaped_len);
+	/*
+	 * By walking backwards in the buffer from the end positions
+	 * of both the original and escaped strings, we avoid using
+	 * extra space. Characters in the original string are
+	 * overwritten only after they have already been escaped.
+	 */
+	str[escaped_len] = '\0';
+	next = &str[*len - 1];
+	dest = &str[escaped_len - 1];
+	while (1) {
```

Comment:
Nitpick, but could be a `for (; next != str; next--)` instead.  Can remove the last few lines in the loop body.